### PR TITLE
Refactor data fetching, move replies, add better tests

### PR DESCRIPTION
**Pull request by schwers**
In current 2X code, async actions that fetch data from the api generally follow this pattern:
```javascript
const fetch = params => async (dispatch, getState) => {
  const id = paramsToId(params);
  const currentRequest = getState().someRequests[id];
  if (currentRequest) { return; }

  dispatch(fetching(id));

  const res = await Endpoint.get(apiOptionsFromState(getState(), params);
  dispatch(receivedApiResponse(res));
  dispatch(received(id, res.results));
};
```

This last bit of dispatching multiple actions isn't quite idiomatic redux and due to a lack of batched actions causes us to run selectors (and possibly re-render) twice. The largest thing this patch does is refactor away `dispatch(receievedAPIResponse(res))`. In doing so some of the naming conventions in action files are made more consistent, the generalized `reducer/apiResponse.js` is removed, in its place are reducers and tests for each specific type of Model (e.g. `posts.js`). 

There's also some logical cleanup of where two actions live: toggling subreddit subscription now lives in `action/subscribedSubreddits.js` with the actions to fetch all of your subscribed subreddits. The action to reply to a post or comment now lives in `actions/reply.js` because even though replies are comments, the action it self acts on comments OR posts. 

Tests are added for the new reducers and updated to reflect the refactor. Note that logged in / logged out tests now all prefill the state of their test's reducer so that we really know log in / log out clears the state to default.

👓 @phil303 @nramadas

### CODE REVIEW

File: src/app/reducers/accountRequests.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import merge from '@r/platform/merge';
+
 import * as accountActions from 'app/actions/accounts';
 import { newAccountRequest } from 'app/models/AccountRequest';
```
Comment (phil303) [inline src/app/reducers/accountRequests.js]: i'm not sure these are even necessary. It just feels like indirection.

File: src/app/reducers/accounts.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+export default function(state=DEFAULT, action={}) {
+  switch (action.type) {
+    case loginActions.LOGGED_IN:
+    case loginActions.LOGGED_OUT: {
+      return DEFAULT;
+    }
+
+    case accountActions.RECEIVED_ACCOUNT: {
+      const { accounts } = action.apiResponse;
+      console.log('accounts are:', accounts);
```
Comment (phil303) [inline src/app/reducers/accounts.js]: print statement

Comment (schwers) [inline src/app/reducers/accounts.js]: shucks, that should really show up in `npm run lint` :/

File: src/app/reducers/accountRequests.js (hunk 1 above)
Comment (schwers) [inline src/app/reducers/accountRequests.js]: Just for my understanding, do you mean the object creation should just live in the reducer? Or something else?

Comment (phil303) [inline src/app/reducers/accountRequests.js]: yea just `return { id: name, loading: true }`

Comment (phil303) [inline src/app/reducers/accountRequests.js]: I don't feel strongly. Maybe just moving the function into the file would ease the feeling of misdirection.

File: src/app/reducers/accounts.test.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
+import createTest from '@r/platform/createTest';
+
+import accounts from './accounts';
+import * as accountActions from 'app/actions/accounts';
+import * as loginActions from 'app/actions/login';
+
+createTest({ reducers: { accounts } }, ({ getStore, expect }) => {
+  describe('accoutns', () => {
```
Comment (phil303) [inline src/app/reducers/accounts.test.js]: nit misspelled accounts

File: src/app/reducers/accounts.test.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    describe('LOGGED_IN and LOGGED_OUT', () => {
+      it('should return default on log ou', () => {
```
Comment (phil303) [inline src/app/reducers/accounts.test.js]: misspelled out

File: src/app/reducers/subreddits.test.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        }));
+
+        const { subreddits } = store.getState();
+        expect(subreddits).to.eql({
+          [SUBREDDIT.uuid]: SUBREDDIT,
+        });
+      });
+    });
+
+    describe('RECEIEVED_SUBSCRIBED_SUBREDDITS', () => {
```
Comment (phil303) [inline src/app/reducers/subreddits.test.js]: misspelled `RECEIVED`

**Comment by phil303** [comment]
Overall I'm really happy with how this came out! Thanks for doing it. I think it'll make the code much easier to follow. Some nitty stuff but :fish:!

**Comment by schwers** [comment]
Okay, thanks for the review. I'll fix the typos now and address the models in another pr today

**Pull request merged**